lix/packages/client/src/github/forkStatus.ts

```diff
@@ -0,0 +1,138 @@
+import isoGit from "../../vendored/isomorphic-git/index.js"
+import { makeHttpClient } from "../git-http/client.js"
+import type { RepoContext, RepoState } from "../openRepository.js"
+import { getMeta } from "../github/getMeta.js"
+
+export async function forkStatus(ctx: RepoContext, state: RepoState) {
+	const {
+		gitUrl,
+		debug,
+		dir,
+		cache,
+		owner,
+		repoName,
+		githubClient,
+		gitProxyUrl,
+	} = ctx
+
+	if (!gitUrl) {
+		throw new Error("Could not find repo url, only github supported for forkStatus at the moment")
+	}
+
+	const { isFork, parent } = (await getMeta(ctx)) as {
+		isFork: boolean
+		parent: { url: string }
+	}
+
+	if (!isFork) {
+		return { error: "could not get fork upstream or repo not a fork" }
+	}
+
+	const forkFs = state.nodeishFs
+
```

Comment:
We could be more specific like this? We don't have to do this just that i got the api right here
```suggestion
	const { error, isFork, parent } = (await getMeta(ctx)) as {
		error: any
		isFork: boolean
		parent: { url: string }
	}

	if (error) {
		return { error: "could not get fork upstream" }
	}

	if (!isFork) {
		return { error: "repo is not a fork" }
	}
```

Reply:
sure